auto_tests/tests/per_series_rolling_average.js

```diff
@@ -0,0 +1,127 @@
+  g.setSelection(3); assert.equal("3: Y: 2.5 Z: 30", Util.getLegend());
+  assert.equal(2, g.getOption("rollPeriod", "Y"));
+
+  seriesOpt["Y"] = { rollPeriod: 3 };
+  seriesOpt["Z"] = { rollPeriod: 2 };
+  g.updateOptions ({ series: seriesOpt });
+  g.setSelection(0); assert.equal("0: Y: 0 Z: 0", Util.getLegend());
+  g.setSelection(1); assert.equal("1: Y: 0.5 Z: 5", Util.getLegend());
+  g.setSelection(2); assert.equal("2: Y: 1 Z: 15", Util.getLegend());
+  g.setSelection(3); assert.equal("3: Y: 2 Z: 25", Util.getLegend());
+  assert.equal(3, g.getOption("rollPeriod", "Y"));
+  assert.equal(2, g.getOption("rollPeriod", "Z"));
+  
+  seriesOpt["Y"] = { rollPeriod: 1 };
+  seriesOpt["Z"] = { rollPeriod: 4 };
+  g.updateOptions ({ series: seriesOpt });
+  g.setSelection(0); assert.equal("0: Y: 0 Z: 0", Util.getLegend());
+  g.setSelection(1); assert.equal("1: Y: 1 Z: 5", Util.getLegend());
+  g.setSelection(2); assert.equal("2: Y: 2 Z: 10", Util.getLegend());
+  g.setSelection(3); assert.equal("3: Y: 3 Z: 15", Util.getLegend());
+  assert.equal(1, g.getOption("rollPeriod", "Y"));
+  assert.equal(4, g.getOption("rollPeriod", "Z"));
+  
+  
+  
+  
+  
+});
+
+it('testDefaultToGlobalRollingAverage', function() {
+  var opts = {
+    width: 480,
+    height: 320,
+	rollPeriod: 2,
+      series: {
+          Y: {
+              rollPeriod: 1 }  
+		  }
+  };
+  var data = "X,Y,Z\n" +
+      "0,0,0\n" +
+      "1,1,10\n" +
+      "2,2,20\n" +
+      "3,3,30\n"
+  ;
+
+  var graph = document.getElementById("graph");
+  var g = new Dygraph(graph, data, opts);
+
+  g.setSelection(0); assert.equal("0: Y: 0 Z: 0", Util.getLegend());
+  g.setSelection(1); assert.equal("1: Y: 1 Z: 5", Util.getLegend());
+  g.setSelection(2); assert.equal("2: Y: 2 Z: 15", Util.getLegend());
+  g.setSelection(3); assert.equal("3: Y: 3 Z: 25", Util.getLegend());
+  assert.equal(1, g.getOption("rollPeriod", "Y"));
+  assert.equal(2, g.getOption("rollPeriod", "Z"));
+  assert.equal(2, g.rollPeriod());
+  
+  var seriesOpt = {};
+  seriesOpt["Y"] = { rollPeriod: 3 };
+  g.updateOptions ({ series: seriesOpt });
+  
+  g.setSelection(0); assert.equal("0: Y: 0 Z: 0", Util.getLegend());
+  g.setSelection(1); assert.equal("1: Y: 0.5 Z: 5", Util.getLegend());
+  g.setSelection(2); assert.equal("2: Y: 1 Z: 15", Util.getLegend());
+  g.setSelection(3); assert.equal("3: Y: 2 Z: 25", Util.getLegend());
+  assert.equal(3, g.getOption("rollPeriod", "Y"));
+  assert.equal(2, g.getOption("rollPeriod", "Z"));
+  assert.equal(2, g.rollPeriod());
+  
+  g.updateOptions ({ rollPeriod: 3 });
+  g.setSelection(0); assert.equal("0: Y: 0 Z: 0", Util.getLegend());
+  g.setSelection(1); assert.equal("1: Y: 0.5 Z: 5", Util.getLegend());
+  g.setSelection(2); assert.equal("2: Y: 1 Z: 10", Util.getLegend());
+  g.setSelection(3); assert.equal("3: Y: 2 Z: 20", Util.getLegend());
+  assert.equal(3, g.getOption("rollPeriod", "Y"));
+  assert.equal(3, g.getOption("rollPeriod", "Z"));
```

Comment:
Not sure I understand "Y" is 3 because it was set previously, "Z" is 3 because the global rollPeriod was updated and "Z" hadn't been set separately. On the example initial strokeWidth : 4 for "Z" then the global StrokeWidth is changed to 1 , leaving "X" at 2